source/server/options_impl.cc

```diff
@@ -48,6 +48,9 @@ OptionsImpl::OptionsImpl(int argc, char** argv, const HotRestartVersionCb& hot_r
                                         std::thread::hardware_concurrency(), "uint32_t", cmd);
   TCLAP::ValueArg<std::string> config_path("c", "config-path", "Path to configuration file", false,
                                            "", "string", cmd);
+  TCLAP::ValueArg<std::string> config_json(
```

Comment:
should we really limit it to json? why not accept yaml too? it wouldn't be readable but then again neither is the json. i'm assuming the use case for this is having a container invocation pass the config in. yaml with escaped newlines would work just as well.

Reply:
Yaml is based on indentation so passing a yaml document as a string seems very error prone.

Reply:
agreed. my thought was that some tool is taking a larger human-readable config file (indented json or yaml) and then passing it in as an escaped string.

Reply:
I could see I use for that; but then the tools might as well generate json.

Reply:
not if the tool is merely reading a file and passing an escaped string. to convert it to json would mean i would have to yaml parse it somehow and then dump it in json format. also to your earlier point yaml can be written like `- foo: [bar: {baz: [bop], qux: [crux]}]`

i guess my main concern is that most v2 documentation is written with yaml in mind, and here we're not going to support it. i'll defer to @htuch on this one if you're not convinced.

Reply:
Since YAML is a JSON superset, if we make this `--config-yaml`, don't we get both JSON and YAML? That should satisfy all conceivable use cases I think.

Reply:
FWIW, adding YAML support if we just s/json/yaml/ is only a line or two of code.

Reply:
Done. This indeed was much simpler that I expected.